File: packages/core/src/xmn/connection-info.ts
/*------------------------------------------------------------------------------
   About      : Represents a connection for client & server.
   
   Created on : Sun Jun 25 2017
   Author     : Raghvendra Varma
   
   Copyright (c) 2017 Mubble Networks Private Limited. All rights reserved.
------------------------------------------------------------------------------*/

import { Mubble }               from '../'
import { Protocol }             from './xmn-core'


export interface ConnectionInfo {

  // Client Public identifier
  shortName       : string    // max four character name (only alpha-numeric)

  // Connection attributes
  protocol        : Protocol
  host            : string    // host name of the server
  port            : number    // port of the server
  url             : string    // /api/getTopics Or connectUrl (for WS)
  headers         : Mubble.uObject<any> // empty for client
  ip              : string    // ip address or host name of the client socket

  // Server fields. Not used by client
  msOffset        : number    // this is inferred by the server based on client's now field. Api/event need not use this
  lastEventTs     : number    // Must be set before an event is processed on server

  customData      : Mubble.uObject<any>
}

File: packages/android/xmn/connection-info.kt
package `in`.mubble.android.xmn

import xmn.ClientIdentity

enum class Protocol {HTTP, WEBSOCKET, HTTPS}

open class ConnectionInfo(val shortName: String, val uniqueId: String) {

  // Connection attributes
  var protocol            : Protocol        = Protocol.WEBSOCKET
  var host                : String          = ""        // host name of the server
  var port                : Int             = 80        // port of the server
  var url                 : String          = ""        // /api/getTopics Or connectUrl (for WS)

  // Information passed by the client: to be used by Xmn internally
  var publicRequest       : Boolean         = false
  var useEncryption       : Boolean         = false

  // Information passed by the client used by
  var location            : String          = ""        // it is in form of serialized json object
  var networkType         : String          = ""
  var clientIdentity      : ClientIdentity? = null

  // provider for this connection (WebSocket, Http etc.)
  var provider            : WsAndroid?      = null      // The protocol provider keeps it's custom data here
  var syncKey             : ByteArray       = ByteArray(32)
}